source/d3d9/runtime_d3d9.hpp

```diff
@@ -94,13 +117,31 @@ namespace reshade::d3d9
 		UINT _num_samplers;
 		UINT _num_simultaneous_rendertargets;
 		bool _disable_intz = false;
+		bool _preserve_depth_buffer = false;
+		bool _disable_depth_buffer_size_restriction = false;
+		int _preserve_starting_index = 0;
+		int _preserve_selected_index = 0;
+		bool _auto_preserve = true;
+		bool _auto_preserve_on_drawcalls = false;
+		bool _auto_preserve_on_vertices = true;
+		bool _multi_depthstencil = false;
+		int _preserve_vertices = 0;
+		int _preserve_drawcalls = 0;
 		bool _is_multisampling_enabled = false;
+		int _clear_buffer_idx = 0;
+		int _clear_idx = 0;
+		int _db_vertices = 0;
+		int _db_drawcalls = 0;
+		int _previous_best_vertices = 0;
+		int _previous_best_drawcalls = 0;
 		D3DFORMAT _backbuffer_format = D3DFMT_UNKNOWN;
 		com_ptr<IDirect3DStateBlock9> _app_state;
 		com_ptr<IDirect3DSurface9> _depthstencil;
 		com_ptr<IDirect3DSurface9> _depthstencil_replacement;
 		com_ptr<IDirect3DSurface9> _default_depthstencil;
 		std::unordered_map<IDirect3DSurface9 *, depth_source_info> _depth_source_table;
+		std::map<int, depth_buffer_info> _depth_buffer_table;
+		std::map<int, depth_clearing_info> _depth_clearing_table;
```

Comment:
Do these have to be `map`s? Looks to me like the clear indices are always linear, so a `vector` would do?